bentoml/_internal/types.py

```diff
@@ -268,6 +268,23 @@ def _format_dt(d: t.Union[datetime, str], _: type) -> datetime:
 cattr.register_structure_hook(datetime, _format_dt)  # type: ignore[misc]
 
 
+class FrozenDict(t.Dict[str, t.Any]):
```

Comment:
This solution actually breaks the consensus of typing. We devs see the arg type is Dict, they will assume that they can do `dict.update` on it.